addons/godot-firebase/auth/auth.gd

```diff
@@ -328,14 +328,23 @@ func manual_token_refresh(auth_data):
 # On an error, this function with emit the signal 'login_failed' and print the error to the console
 func _on_FirebaseAuth_request_completed(result : int, response_code : int, headers : PoolStringArray, body : PoolByteArray) -> void:
     is_busy = false
-    var bod = body.get_string_from_utf8()
-    var json_result = JSON.parse(bod)
-    if json_result.error != OK:
-        Firebase._printerr("Error while parsing auth body json")
-        emit_signal("auth_request", "Error while parsing auth body json", json_result)
-        return
-        
-    var res = json_result.result
+    var res
```

Comment:
This type of thing is absolutely fine and great for testing, but we shouldn't leave it in the actual release. We should probably drop this commit, or remove this change specifically.

Reply:
Happy to take input on another way to address this; intent of the last commit is to ensure even without network connection, the right signals are still emitted so it can be handled, as the current master version fails with no signal emission.

Is your comment over having this standalone `var res` defined without assignment/type? Or the other changes in this more recent commit?

A broader thought: I wonder for consistency if it would be good to change how this function handles parsing errors (which is why `var` is on its own right now). Right now it just returns early (and with my contribution as is, emits the `auth_request` signal). However, in the `if response_code .. else` block, for non 200 responses, the function attempts to raise the corresponding signal(s) based on what request was being made. Should we update a parsing failure to do the same, so that we can always depend on *either* the success or corresponding failure signal to be emitted based on the request made? (Where the `auth_request` signal always emit and is equivalent to the `finally` of a `try/except/finally` clause).

Happy to break into another PR if it's expanding too much.

Reply:
Sorry, I didn't make that clear - wasn't just the var. It was the set of changes here and below. My understanding is when 0 is returned, there's no error - is that not the case?

Reply:
No error would be HTTP response_code == 200 (ie `HTTPClient.RESPONSE_OK`). In the scenario where there's no internet connection, there actually isn't a response code at all, because you weren't able to reach the server (where normally a server error would be something in the 500's, or a wrong path would be in the 400's). It happens to be that Godot's implementation of no response is to assign a value of zero. Maybe there's an appropriate `HTTPClient.RESPONSE` code for that, but I didn't see an obvious fit when I scrolled through the list.

Reply:
Ohhh, you were looking at the response code. I was thinking with was the normal Godot global.Error value, not a response code. Okay, then yeah, I don't see anything wrong with this!